newbank/server/Customer.java

```diff
@@ -36,6 +36,21 @@ public String depositToAccount(String account, double amount) {
 	return "FAIL";
 }
 
+	public String withdrawFromAccount(String account, double amount) {
+		for(Account a : accounts) {
+			if(a.getAccountBalance() - amount < 0) {
+					return "Insufficient funds";}
+			else {
+				if(a.getAccountName() == account){
+					a.withdraw(amount);
+
+					return "Withdrawal successful! \n" + a.toString();
+				}
+			}
+			}
+		return "FAIL";
+	}
```

Comment:
I think there's a mistake in this method. I think you need to do the check for whether it's the right account *before* doing the check for whether there are sufficient funds. Say I've got 2 accounts:
1. Main - £100
2. Savings - £1470

and I ask to withdraw £250 from savings.

If the Main account happens to be the first in the for loop, your method will reject my request, because Main has insufficient funds - even though Savings has enough.

Does that make sense?